flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -335,6 +341,19 @@ public void alterTable(ObjectPath tablePath, CatalogBaseTable newTable, boolean
     throw new UnsupportedOperationException("Not support alterTable now.");
   }
 
+  CatalogLoader getCatalogLoader() {
+    return catalogLoader;
+  }
+
+  Configuration getHadoopConf() {
+    return hadoopConf;
+  }
+
+  @Override
+  public Optional<TableFactory> getTableFactory() {
```

Comment:
Q: The javadoc says it's deprecated now,  What's  the time for us to use the `getFactory` in future ? 

```java
	 * @deprecated Use {@link #getFactory()} for the new factory stack. The new factory stack uses the
	 *             new table sources and sinks defined in FLIP-95 and a slightly different discovery mechanism.
	 */
	@Deprecated
	default Optional<TableFactory> getTableFactory() {
```

Reply:
Until the new API is really ready...
In the 1.11 and master, FLIP-95 interfaces still lack many things. I am not sure about Flink 1.12, maybe 1.13 is the time.